File: backend/InventoryManagementApi/src/main/java/org/ngcvfb/inventorymanagementapi/model/Product.java
package org.ngcvfb.inventorymanagementapi.model;

import jakarta.persistence.*;
import java.math.BigDecimal;

@Entity
@Table(name = "products")
public class Product {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false)
    private String sku;

    @Column(nullable = false)
    private String name;

    private String description;

    private BigDecimal price = BigDecimal.ZERO;

    private Long currentStock = 0L;

    private String imagePath;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "category_id")
    private Category category;

    @Version
    private Long version;

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getSku() { return sku; }
    public void setSku(String sku) { this.sku = sku; }

    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    public BigDecimal getPrice() { return price; }
    public void setPrice(BigDecimal price) { this.price = price; }

    public Long getCurrentStock() { return currentStock; }
    public void setCurrentStock(Long currentStock) { this.currentStock = currentStock; }

    public String getImagePath() { return imagePath; }
    public void setImagePath(String imagePath) { this.imagePath = imagePath; }

    public Category getCategory() { return category; }
    public void setCategory(Category category) { this.category = category; }

    public Long getVersion() { return version; }
    public void setVersion(Long version) { this.version = version; }
}

File: frontend/lib/models/product.dart
class Product {
  final int id;
  final String name;
  final String sku;
  final int currentStock;
  final double price;
  final String? imagePath;
  final int? categoryId;
  final String? categoryName;

  Product({
    required this.id,
    required this.name,
    required this.sku,
    required this.currentStock,
    required this.price,
    this.imagePath,
    this.categoryId,
    this.categoryName,
  });

  String get imageUrl => imagePath ?? '';

  Product copyWith({
    int? id,
    String? name,
    String? sku,
    int? currentStock,
    double? price,
    String? imagePath,
    int? categoryId,
    String? categoryName,
  }) {
    return Product(
      id: id ?? this.id,
      name: name ?? this.name,
      sku: sku ?? this.sku,
      currentStock: currentStock ?? this.currentStock,
      price: price ?? this.price,
      imagePath: imagePath ?? this.imagePath,
      categoryId: categoryId ?? this.categoryId,
      categoryName: categoryName ?? this.categoryName,
    );
  }

  factory Product.fromJson(Map<String, dynamic> json) {
    final category = json['category'] as Map<String, dynamic>?;
    return Product(
      id: json['id'],
      name: json['name'],
      sku: json['sku'],
      currentStock: json['currentStock'],
      price: (json['price'] as num).toDouble(),
      imagePath: json['imagePath'],
      categoryId: category?['id'],
      categoryName: category?['name'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'sku': sku,
      'description': '',
      'currentStock': currentStock,
      'price': price,
      'imagePath': imagePath,
      'categoryId': categoryId,
    };
  }
}
